apps/studio/src/contexts/EditorDrawerContext.tsx

```diff
@@ -61,6 +72,51 @@ export function EditorDrawerProvider({
   // Holding state for images/files that have been modified in the page
   const [modifiedAssets, setModifiedAssets] = useState<ModifiedAsset[]>([])
   const [addedBlockIndex, setAddedBlockIndex] = useState<number | null>(null)
+  const toast = useToast({ status: "error" })
+  const utils = trpc.useUtils()
+  const { mutate: insertChildpageBlock } =
+    trpc.folder.insertChildpageBlock.useMutation({
+      onSuccess: async () => {
+        await utils.page.readPage.invalidate({ pageId, siteId })
+        await utils.page.readPageAndBlob.invalidate({ pageId, siteId })
+      },
+      onError: (error) => {
+        toast({
+          title: "Failed to update blocks",
+          description: error.message,
+          ...BRIEF_TOAST_SETTINGS,
+        })
+      },
+    })
+
+  useEffect(() => {
```

Comment:
question: might be lacking a bit of context here so for my understanding, this `useEffect` is needed because we want to ensure that if the user is editing an index page, they would be able to have access to the child pages block?

I assume this is so that we can keep things backward compatible without having to run a migration of existing index pages to add this child pages block?

^ if this is so, then actually why not just do the migration directly so we don't need to keep this part of the code?

Reply:
1. we can't do this migration before-hand because it'll result in studio showing an unknown block
2. if we do the migration after, we'd need to comms out to users so that they'd avoid data corruption (ie, i do the migration whilst they're updating, resulting in studio rejecting the update cos the stored state isn't equivalent to the user's state).
3. if we run the migration at night **to avoid impacting users**, there will be a time lag where the children pages aren't generated on end sites - because of this, we'd need the intermediate state between feature release and migration

Reply:
Hmm sorry might not be understanding, I thought that adding the childpage block is backward compatible such that if the block is not there, then we still generate it on the components' side. If the block is there, then no additional generation necessary. In both cases, it should be transparent to the user?

We can introduce the component first so Studio knows about it, then later on perform the migration to add the childpages block to existing index pages.

Reply:
[resolved offline] - we're keeping this `useEffect` temporarily whilst we add the migration, then we'll remove it afterwards